core/field_number.js

```diff
@@ -270,29 +296,25 @@ class FieldNumberPicker extends Blockly.FieldNumber {
     e.preventDefault();
   }
 
-  doClassValidation_(
-    newValue
-  ) {
-    if (newValue === '') return newValue;
-    return super.doClassValidation_(newValue);
+  doClassValidation_(newValue) {
```

Comment:
hrmm I remember there being some issue about colour fields having injection vulnerabilities before we started consistently doing validation. This means that if folks call `setValue` or load malformed XML it will no longer be validated. I don't know much of anything about security, so just want to check that this lgty.

Reply:
It looks like that was based on code generation which Scratch doesn't do, and in general relying on sanitization is a risky approach, but on a second look it seems like we can allow the default validation, and it actually is helpful in ensuring that non-numeric allowed characters parse correctly; I think I may have made this change in an earlier in-flight state. Reverting accordingly.